resources/product-listing.js

```diff
@@ -19,7 +19,38 @@ function ProductListing(shopify) {
   this.key = 'product_listing';
 }
 
-assign(ProductListing.prototype, omit(base, ['create', 'update']));
+assign(ProductListing.prototype, omit(base, ['create', 'update', 'buildUrl', 'productIds', 'count', 'delete']));
+
+/**
+ * Builds the request URL.
+ *
+ * @param {Number|String} [id] Record ID
+ * @param {Object} [query] Query parameters
+ * @return {Object} URL object
+ * @private
+ */
+ProductListing.prototype.buildUrl = function(id, query) {
```

Comment:
I know this follows my suggestion but on second thought I think it's easier to override only the methods that need the special header instead of the `buildUrl()`. It should keep things a little cleaner.